src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -372,39 +381,87 @@ private InputStream createXsltAsInputStreamIfPresent(MultipartFile xslt) {
         return new ByteArrayInputStream(new byte[0]);
     }
 
-    private static class CountryView {
-
-        @JsonProperty("name")
-        private final String name;
-        @JsonProperty("xmlValue")
-        private final String xmlValue;
-
-        /**
-         * Instantiates a new Country view.
-         *
-         * @param country the country
-         */
-        CountryView(Country country) {
-            this.name = country.name();
-            this.xmlValue = country.xmlValue();
+    private CompressedFileExtension checkIfFileTypeFromUrlIsValid(String url) {
+
+        try {
+            if (url.startsWith("file:/")) {
+                Path path = Path.of(url);
+                String fileContentType = Files.probeContentType(path);
+
+                return checkIfIsValidFileType(fileContentType);
+            } else {
+
+                URLConnection urlConnection = new URL(url).openConnection();
+                String fileContentType = urlConnection.getContentType();
+
+                if (StringUtils.isEmpty(fileContentType)) {
+                    throw new InvalidCompressedFileException(new Exception("There was an issue inspecting file's content type"));
+                }
+
+                return checkIfIsValidFileType(fileContentType);
+
+
+            }
+        } catch (IOException e) {
+            throw new InvalidCompressedFileException(new Exception("There was an issue while extracting file from url"));
+
         }
     }
 
-    private static class LanguageView {
-
-        @JsonProperty("name")
-        private final String name;
-        @JsonProperty("xmlValue")
-        private final String xmlValue;
-
-        /**
-         * Instantiates a new Language view.
-         *
-         * @param language the language
-         */
-        LanguageView(Language language) {
-            this.name = language.name();
-            this.xmlValue = language.xmlValue();
+        private CompressedFileExtension checkIfFileTypeFromFileHarvestIsValid (MultipartFile uploadedFile){
```

Comment:
My suggestion would be
a) to split the method `checkIfFileTypeFromFileHarvestIsValid` in two parts 
```
private void checkArgumentMultipartFile(MultipartFile uploadedFile) {
String fileContentType = uploadedFile.getContentType();
            if (StringUtils.isEmpty(fileContentType) || 
                  !(fileContentType.contains("x-tar") || fileContentType.contains("zip") || fileContentType.contains("gzip"))) {
                throw new InvalidCompressedFileException(new Exception("Content type file is not supported"));
            }
}  
```
and a separated function just for getting the type
```
CompressedFileExtension getFileContentType(MultipartFile uploadedFile) {
           if (uploadedFile.getContentType().contains("gzip")) {
                return CompressedFileExtension.TAR_GZ;
            } else if (uploadedFile.getContentType().contains("zip")) {
                return CompressedFileExtension.ZIP;
            } else if (uploadedFile.getContentType().contains("x-tar")) {
                return CompressedFileExtension.TAR;
}
```
call the method would look like this...
```
 harvestPublishService.runHarvestFileAsync(dataset, datasetMetadata, getFileContentType(dataset))
                .exceptionally(e -> datasetLogService.logException(createdDatasetId, e));

```

Reply:
the same idea would apply to the url link validation... let me know what are your thoughts. also another option could be renaming `checkIfFileTypeFromFileHarvestIsValid` to something that says that is doing the two things at the same time.

Reply:
With your suggestion it seems that we are repeating the step of checking the what content type do we have.
According to your suggestion, we would do it first for checking if it's valid and again later to see which type of `CompressedFileExtension` to return.
I prefer to keep the way I currently have because it avoids this repetition. What I will do is change the name of the methods so they are not misleading

Reply:
:+1: